src/@[scope]/[name].md.js

```diff
@@ -0,0 +1,276 @@
+        <div class="green">${recentStargazerCount.toLocaleString("en-US", {signDisplay: "always"})} in 7d</div>
+      </div>
+    </div>
+  </a>
+  <a class="card">
+    <h2>Weekly downloads</h2>
+    <div style="display: flex; flex-wrap: wrap; gap: 0.5rem; align-items: baseline;">
+      <div class="big">${weeklyDownloadsCount.toLocaleString("en-US")}</div>
+      <div class="${weeklyDownloadsChange > 0 ? "green" : weeklyDownloadsChange < 0 ? "red" : "muted"}">${weeklyDownloadsChange.toLocaleString("en-US", {style: "percent", signDisplay: "always"})}</div>
+    </div>
+  </a>
+  <a href=https://github.com/${githubRepo}/releases class="card">
+    <h2>Latest release</h2>
+    <div style="display: flex; flex-wrap: wrap; gap: 0.5rem; align-items: baseline;">
+      <div class="big">${npmInfo["dist-tags"].latest}</div>
+      <div class="muted">${utcDay.count(new Date(npmInfo.time[npmInfo["dist-tags"].latest]), today).toLocaleString("en-US")} days ago</div>
+    </div>
+  </a>
+</div>
+
+<div class="grid grid-cols-1">
+  <div class="card">
+    <h2>Daily downloads</h2>
+    <h3>28d <b style="color: var(--theme-foreground);">—</b> and 7d <b style="color: var(--theme-foreground-focus);">—</b> average</h3>
+    <div style="min-height: 390px;">$\{Plot.plot({
+      width,
+      height: 400,
+      marginLeft: 0,
+      marginRight: 60,
+      x: {domain},
+      y: {label: "downloads", domain: [0, d3.quantile(downloads, 0.995, (d) => d.value)]},
+      marks: [
+        Plot.axisY({anchor: "right", label: null}),
+        Plot.areaY(downloads, {x: "date", y: "value", fillOpacity: 0.2, curve: "step"}),
+        Plot.ruleY([0]),
+        Plot.lineY(downloads, Plot.windowY({k: 7, anchor: "start", strict: true}, {x: "date", y: "value", strokeWidth: 1, stroke: "var(--theme-foreground-focus)", curve: "step"})),
+        Plot.lineY(downloads, Plot.windowY({k: 28, anchor: "start", strict: true}, {x: "date", y: "value", stroke: "var(--theme-foreground)", curve: "step"})),
+        Plot.textX(versions.filter((d) => d.date >= start), {x: "date", text: "version", href: (d) => \`https://github.com/${githubRepo}/releases/tag/v$\{d.version\}\`, target: "_blank", rotate: -90, frameAnchor: "top-right", lineAnchor: "bottom", dx: -4}),
+        Plot.ruleX(versions.filter((d) => d.date >= start), {x: "date", strokeOpacity: 0.2}),
+        Plot.tip(downloads, Plot.pointerX({x: "date", y: "value"}))
+      ]
+    })}</div>
+  </div>
+</div>
+
+<div class="grid grid-cols-1">
+  <div class="card">
+    <h2>Downloads by version</h2>
+    <h3>Last seven days; top 10 versions</h3>
+    <div>$\{Plot.plot({
+      width,
+      label: null,
+      marginLeft: 40,
+      marginRight: 60,
+      x: {axis: "top", grid: true},
+      marks: [
+        Plot.barX(versions.filter((d) => d.downloads > 0), {
+          y: "version",
+          x: "downloads",
+          sort: {y: "x", reverse: true, limit: 10}
```

Comment:
I'd prefer sorting versions by semver rather than by downloads… otherwise all the charts are showing about the same (power law?) distribution. It's more interesting to see a different pattern if, say, there is a recent version, or if an older version has had a huge number of downloads that the more recent can't match. wdyt?

Reply:
The problem is that the **limit** option is controlled by the same order, so if I use semver to show the latest 10 versions, it won’t necessarily include the top 10 versions. For example with Plot, 0.4.3 is in the top 10 versions, but not in the latest 10 versions. I can compute the y-domain manually though…

Reply:
Okay, done. Please that this works…

```js
import semverCompare from "npm:semver/functions/compare";
```

Reply:
This was a nice suggestion. I like how it makes it more obvious when an old release is popular.

<img width="951" alt="Screenshot 2024-10-18 at 4 11 13 PM" src="https://github.com/user-attachments/assets/e695b349-e182-46e2-a977-de8abc5bf068">

Now I wish I knew _why_ the old release is popular. 🙂